src/components/shared/TextField/index.tsx

```diff
@@ -21,20 +22,32 @@ const TextField = ({
   maxLength,
   onChange,
   onBlur,
-  value
+  value,
+  match
 }: TextFieldProps) => {
   const inputClasses = classnames(
     'usa-input',
     { 'easi-textfield--disabled': disabled },
     { 'usa-input--error': error }
   );
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (match) {
```

Comment:
Adding a `match` prop which takes in a regex. It is optional, but it will allow us to only capture characters a field allows.